internal/oauth2/lazy_token_test.go

```diff
@@ -61,23 +46,11 @@ func TestLazyToken(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "mock-token-1", token)
 
-	// ensure no token is not part of the logs
-	field := collector.Last()["msg"]
-	assert.Equal(t, 1, len(collector.All()))
-	assert.Contains(t, field, "Acquired new token")
-	assert.NotContains(t, field, "mock-token-1")
-	collector.Reset()
-
+	// reuses the token when it's not expired
 	token, err = lazyToken.Token(ctx)
 	require.NoError(t, err)
 	require.Equal(t, "mock-token-1", token)
 
-	field = collector.Last()["msg"]
-	assert.Equal(t, 1, len(collector.All()))
-	assert.Contains(t, field, "AccessToken reused")
-	assert.NotContains(t, field, "mock-token-1")
-	collector.Reset()
-
```

Comment:
Dropped all log-related testing, the test itself is good it checks via counter if tokens are reused or not.